utils.js

```diff
@@ -0,0 +1,29 @@
+const pickEditSchema = (req, res, next) => {
+    const question = require(`./questions/${req.params.id}.js`)
+    return checkSchema(question.schema)[0](req, res, next)
+}
+/*
+  original format is an array of error objects: https://express-validator.github.io/docs/validation-result-api.html
+  convert that to an object where the key is the parameter name and value is the error object
+  ie,
+  [
+    { param: 'name', msg: 'Cannot be empty', ... },
+    { param: 'number', msg: 'Cannot be empty', ... }
+  ]
+  to
+  {
+    name: { param: 'name', msg: 'Cannot be empty', ... },
+    number: { param: 'number', msg: 'Cannot be empty', ... }
+  }
+*/
+const errorArray2ErrorObject = (errors = []) => {
+    return errors.array({ onlyFirstError: true }).reduce((map, obj) => {
```

Comment:
Does `onlyFirstError: true` mean that if I mess up 2 fields I only get shown the first error? If so, is this what users expect? (Personally, I really don't know)

Reply:
It's more for cases where multiple validations fail for one field.

Like if you wanted to say:
- sin must be not empty
- sin must be 9 characters
- sin can only be numbers

If I entered "SIN", then i would get back 2 errors: "9 chars" and "only numbers". Asking for just the first error message would only display "9 characters" and then that's what would show up on the form.

I think for our current purposes, this is fine.